plugins/zstack_stage.py

```
@@ -0,0 +1,276 @@
+            filename = f"{config.last_path}/{basename}{config.last_extension}"
+
+            logging.info("Saving stage z-stack data to: %s", filename)
+
+            # Find appropriate exporter and save
+            exporter = dataio.find_fittest_converter(filename)
+            exporter.export(filename, data)
+
+            logging.info("Data saved successfully to: %s", filename)
+
+        except Exception as e:
+            logging.exception("Failed to save data: %s", e)
```

Ruff (0.15.1)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)